### PR TITLE
#95 Rename Batch to GlobalWindowing

### LINKED ISSUE #95: Consider aligning naming of "batch windowing"
Apache Beam as well as Apache Flink refer to what we named `Batch` and `Batch.BatchWindow` as `GlobalWindows` and `GlobalWindow` respectively. I think the term "global" makes more sense. 1) Windowing is a separate concept/value to "batch/stream processing". 2) By aligning the names we create less confusion for new users.

In particular, I'd like to propose renaming:

* `Batch` => `GlobalWindowing`
* `Batch.BatchWindow` => `GlobalWindowing.GlobalWindow` or just `GlobalWindowing.Window`

Please comment if renaming doesn't make sense to you.

### PR DESCRIPTION
**Pull request by xitep**
* Resolves #95

### CODE REVIEW

File: euphoria-core/src/main/java/cz/seznam/euphoria/core/client/dataset/windowing/GlobalWindowing.java [hunk 1]
```diff
@@ -50,27 +53,27 @@ private Object readResolve() throws ObjectStreamException {
     }
 
     @Override
-    public int compareTo(BatchWindow o) {
+    public int compareTo(Window o) {
```
Comment (je-ik) [inline euphoria-core/src/main/java/cz/seznam/euphoria/core/client/dataset/windowing/GlobalWindowing.java]: This seems to be non conformant with the `equals` method. Is it on purpose?

File: euphoria-core/src/main/java/cz/seznam/euphoria/core/client/dataset/windowing/GlobalWindowing.java [hunk 2]
```diff
@@ -50,27 +53,27 @@ private Object readResolve() throws ObjectStreamException {
...
-  private final static Batch<?> INSTANCE = new Batch<>();
-  private Batch() {}
+  private final static GlobalWindowing<?> INSTANCE = new GlobalWindowing<>();
+  private GlobalWindowing() {}
 
   @Override
-  public Iterable<BatchWindow> assignWindowsToElement(WindowedElement<?, T> el) {
-    return Collections.singleton(BatchWindow.INSTANCE);
+  public Iterable<Window> assignWindowsToElement(WindowedElement<?, T> el) {
+    return Collections.singleton(Window.INSTANCE);
```
Comment (je-ik) [inline euphoria-core/src/main/java/cz/seznam/euphoria/core/client/dataset/windowing/GlobalWindowing.java]: Here is possible a micro optimization - we could create the set and store it in static field. It should be immutable, so it seems safe.

File: euphoria-inmem/src/main/java/cz/seznam/euphoria/inmem/InMemExecutor.java [hunk 3]
```diff
@@ -712,7 +711,7 @@ private InputProvider execReduceStateByKey(
                   (oldVal, newVal) -> oldVal < newVal ? newVal : oldVal);
               // determine partition
               Object key = keyExtractor.apply(datum.getElement());
-              final Iterable<Window> targetWindows;
+              final Iterable<cz.seznam.euphoria.core.client.dataset.windowing.Window> targetWindows;
```
Comment (je-ik) [inline euphoria-inmem/src/main/java/cz/seznam/euphoria/inmem/InMemExecutor.java]: Why we dropped the import here and are referencing the class by whole name?

File: euphoria-core/src/main/java/cz/seznam/euphoria/core/client/dataset/windowing/GlobalWindowing.java (hunk 1 above)
Comment (xitep) [inline euphoria-core/src/main/java/cz/seznam/euphoria/core/client/dataset/windowing/GlobalWindowing.java]: hm, i think it is. `Window` here refers here to `GlobalWindowing.Window`; i'll make that explicit.

File: euphoria-core/src/main/java/cz/seznam/euphoria/core/client/dataset/windowing/GlobalWindowing.java (hunk 2 above)
Comment (xitep) [inline euphoria-core/src/main/java/cz/seznam/euphoria/core/client/dataset/windowing/GlobalWindowing.java]: yeah, why not. seem safe to me as well.

File: euphoria-inmem/src/main/java/cz/seznam/euphoria/inmem/InMemExecutor.java (hunk 3 above)
Comment (xitep) [inline euphoria-inmem/src/main/java/cz/seznam/euphoria/inmem/InMemExecutor.java]: oops ... will fix that; thanks :+1:

File: euphoria-core/src/main/java/cz/seznam/euphoria/core/client/dataset/windowing/GlobalWindowing.java (hunk 1 above)
Comment (je-ik) [inline euphoria-core/src/main/java/cz/seznam/euphoria/core/client/dataset/windowing/GlobalWindowing.java]: Aj, you're right, my fault! :)

**Pull request merged**